core/App/ehsm_core_test.cpp

```diff
@@ -42,577 +42,60 @@
-        goto cleanup;
+        goto out;  
     }
+    printf("NAPI_CreateKey Json = %s\n", returnJsonChar);
     printf("Create CMK with AES-128 SUCCESSFULLY!\n");
-    dump_data(cmk.keyblob, cmk.keybloblen);
-
-    plaintext.datalen = 16;
-    plaintext.data = (uint8_t*)malloc(plaintext.datalen);
-    if (plaintext.data == NULL) {
-        ret = EH_DEVICE_MEMORY;
-        goto cleanup;
-    }
-    memset(plaintext.data, 'A', plaintext.datalen);
-    dump_data(plaintext.data, plaintext.datalen);
-
-    aad.datalen = 0;
-    aad.data = NULL;
-    ciphertext.datalen = 0;
-    ret = Encrypt(&cmk, &plaintext, &aad, &ciphertext);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to get data size of Encrypt!\n", ret);
-        goto cleanup;
-    }
 
-    ciphertext.data = (uint8_t*)malloc(ciphertext.datalen);
-    if (ciphertext.data == NULL) {
-        ret = EH_DEVICE_MEMORY;
-        goto cleanup;
-    }
+    char* cmk_base64;
+    cmk_base64 = retJsonObj.parseStringData(returnJsonChar, "cmk_base64");
 
-    ret = Encrypt(&cmk, &plaintext, &aad, &ciphertext);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to Encrypt the plaittext data\n", ret);
-        goto cleanup;
+    returnJsonChar = NAPI_Encrypt(cmk_base64, plaintext, aad);
+    if(returnJsonChar == nullptr){
+        printf("Failed to Encrypt the plaittext data\n");
+        goto out; 
     }
+    printf("NAPI_Encrypt json = %s\n", returnJsonChar);
     printf("Encrypt data SUCCESSFULLY!\n");
-    dump_data(ciphertext.data, ciphertext.datalen);
 
-    plaintext2.datalen = 0;
-    ret = Decrypt(&cmk, &ciphertext, &aad, &plaintext2);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to get data size of Decrypt!\n", ret);
-        goto cleanup;
-    }
-    plaintext2.data = (uint8_t*)malloc(plaintext2.datalen);
-    if (plaintext2.data == NULL) {
-        ret = EH_DEVICE_MEMORY;
-        goto cleanup;
-    }
+    char* ciphertext_base64;
+    ciphertext_base64 = retJsonObj.parseStringData(returnJsonChar, "ciphertext_base64");
 
-    ret = Decrypt(&cmk, &ciphertext, &aad, &plaintext2);
-    if (ret != EH_OK) {
-        printf("Failed(%d) to Decrypt the data\n", ret);
-        goto cleanup;
+    returnJsonChar = NAPI_Decrypt(cmk_base64, ciphertext_base64, aad);
+    if(returnJsonChar == nullptr){
+        printf("Failed to Decrypt the data\n");
+        goto out; 
     }
+    printf("NAPI_Decrypt json = %s\n", returnJsonChar);
+    char* plaintext_base64;
+    plaintext_base64 = retJsonObj.parseStringData(returnJsonChar, "plaintext_base64");
+    printf("plaintext = %s\n",base64_decode(plaintext_base64).c_str());
     printf("Decrypt data SUCCESSFULLY!\n");
-    dump_data(plaintext2.data, plaintext2.datalen);
-cleanup:
-    SAFE_FREE(cmk.keyblob);
-    SAFE_FREE(plaintext.data);
-    SAFE_FREE(ciphertext.data);
-    SAFE_FREE(plaintext2.data);
-
-    printf("============testAES Done==========\n");
-
-    return ret;
-}
-
-ehsm_status_t testRSA()
```

Comment:
don't delete the original test cases unless you have already implemented the new one.